spec/resources/api/v2/plate_creation_resource_spec.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require './app/resources/api/v2/plate_creation_resource'
+
+RSpec.describe Api::V2::PlateCreationResource, type: :resource do
+  subject(:resource) { described_class.new(resource_model, {}) }
+
+  let(:resource_model) { build_stubbed(:plate_creation) }
+
+  # Model Name
+  it { is_expected.to have_model_name 'PlateCreation' }
+
+  # Attributes
+  it { is_expected.to have_writeonly_attribute :child_purpose_uuid }
+  it { is_expected.to have_writeonly_attribute :parent_uuid }
+  it { is_expected.to have_writeonly_attribute :user_uuid }
+  it { is_expected.to have_readonly_attribute :uuid }
+
+  # Relationships
+  it { is_expected.to have_a_readonly_have_one(:child, 'Plate') }
+  it { is_expected.to have_a_writable_have_one(:child_purpose, 'PlatePurpose') }
+  it { is_expected.to have_a_writable_have_one(:parent, 'Plate') }
+  it { is_expected.to have_a_writable_have_one(:user, 'User') }
+end
```

Comment:
@StephenHulme 

> Does this specificy rule out children that are tubes or tube-racks?

This is the plate_creation endpoint, so it is intended to create plates.  I don't know why this was built so specifically that way, but there it is.

> To confirm, write-only is set but not read, readonly is read but not set, and writable is is read and write?

Not quite true:

- Write-only is set only when the resource is created, and not when you update (PATCH) it.
- Read-only is readable but can never be set via the API.
- Writable is able to be written in both POST and PATCH requests, but says nothing about the readability.

The `have_one` and `have_many` matchers only check for readonly or writable relationships because that's how we've been using them.  In the matchers themselves, you can see that we do checks like `have_attribute` which is from an external library and actually means that the attribute is readable and not that it was defined.  You can define the attribute as readonly and the `have_attribute` returns `false`.  I've tried to be more explicit in my naming.